python/dgl/dataloading/pytorch/__init__.py

```diff
@@ -137,17 +137,20 @@ def collate(self, items):
             return input_nodes, pair_graph, neg_pair_graph, blocks
 
 class _NodeDataLoaderIter:
-    def __init__(self, node_dataloader):
+    def __init__(self, device, node_dataloader):
+        self.device = device
         self.node_dataloader = node_dataloader
         self.iter_ = iter(node_dataloader.dataloader)
 
     def __next__(self):
         input_nodes, output_nodes, blocks = next(self.iter_)
         _restore_blocks_storage(blocks, self.node_dataloader.collator.g)
+        blocks = [blc.to(self.device) for blc in blocks]
```

Comment:
does this run in the sampler process or the trainer process?

Reply:
@aksnzhy said it's in the sampler process.

Reply:
For single-machine training, this should be in the trainer (or main) process?  The sampler process does not (and should not) return a graph with node/edge features.  The main process is responsible for getting the features.

For distributed training, I'm not sure.  Perhaps the sampler process is responsible for getting both the node/edge features and the graph and sending them to the trainer.